### DISCUSSION TITLE
Implementing OTP Verification and Interceptors in Serverpod

### DISCUSSION THREAD

**Discussion by Arnooodles** [serverpod/serverpod#2406] [Q&A]
1. How can I implement SMS/Email OTP verification after calling an endpoint in Serverpod? Is this possible, and if so, is there any guide on implementing this?

2. How can I implement login via phone number with OTP verification in Serverpod? Is there a guide on how to set up a login system where users enter their phone numbers and receive an OTP via SMS/Email before they can proceed with logging in?

3. Can I implement interceptors in Serverpod similar to dio/chopper for OTP verification?  I believe interceptors can be used to catch calls that need OTP verification. Is this possible in Serverpod, and if so, how can it be done?

Note:
I understand that these features require some SMS/SMTP provider. However, for development purposes, is it possible to mock the OTP to a fixed code like "123456" as the valid code?

**Comment by omartarek003** [comment]
Hello, @Arnooodles

Can you tell me how you did it ?
